equistore-core/CMakeLists.txt

```diff
@@ -29,7 +29,9 @@ project(equistore
     LANGUAGES C # we need to declare a language to access CMAKE_SIZEOF_VOID_P later
 )
 
-option(BUILD_SHARED_LIBS "Build a shared library instead of a static one" ON)
+option(BUILD_SHARED_LIBS "Use a shared library by default instead of a static one" ON)
+option(EQUISTORE_INSTALL_BOTH_STATIC_SHARED "Install both shared and static libraries" ON)
```

Comment:
So the reason for having both variable setup like this is 

- BUILD_SHARED_LIBS is the standard CMake way of dealing with "I want a static/shared library"
- cargo always builds both anyway
- the user might want to only install one of them (e.g. inside the Python wheels we only want the shared lib)

Since cargo always build both, we use `BUILD_SHARED_LIBS` to decide whether the `equistore` target is an alias of `equistore::static` or `equistore::shared`.


I'll add a comment explaining what's going on a bit more here.